### PR TITLE
Did the stuff

### PR DESCRIPTION
**Pull request by warnerjon12**

### CODE REVIEW

**Comment by warnerjon12** [comment]
I'm not sure why the one commit has a timestamp that's wrong, but I don't think it really matters

File: src/fifo.c [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
 void fifo_worker_handler(QueueHandle_t requests, QueueHandle_t results, int id) {
-    // TODO
+    struct request_msg data;
+    for (;;) {
+        if (xQueueReceive(requests, &data, portMAX_DELAY) == pdTRUE) {
+            data.output = data.input + 5;
+            data.handled_by = id;
+            if (xQueueSendToBack(results, &data, portMAX_DELAY) != pdTRUE) {
+                printf("Failed to send data:\n\tInput: %d\n\tOutput: %d\n\tHandled by: %u\n",
```
Comment (akcase18) [inline src/fifo.c:10]: I like this print statement

**Review by akcase18 — COMMENTED** [review]
Looks good!

**Review by akcase18 — APPROVED** [review]

**Review by KirraKotsenburg — APPROVED** [review]
Everything looks ready for the merge!

**Pull request merged**